### PR TITLE
Remove support for non-batching servers in request.rs, added tests

### PR DESCRIPTION
**Pull request by pdoud2124**
### Pull Request checklist ###

- [x] **Quality**: This PR builds and tests run cleanly
  - `automation/all_tests.sh` runs to completion and produces no failures
  - Note: For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/master/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

We combed the file for any function supporting non-batching servers and removed the code. We then added a series of tests at the bottom of the file to test multiple records and multiple bytes for one batch along with testing error cases.

### CODE REVIEW

File: .project [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ml8) [inline .project:1]: Remove this file (and the eclipse one) from the PR

File: components/sync15/src/request.rs [hunk 2]
```diff
@@ -24,7 +24,10 @@ struct LimitTracker {
     cur_records: usize,
 }
 
+// The implementation for the limit, allowing for a new limit,
```
Comment (ml8) [inline components/sync15/src/request.rs]: Remove, since the comment on LimitTracker describes its purpose.

File: components/sync15/src/request.rs [hunk 3]
```diff
@@ -34,21 +37,25 @@ impl LimitTracker {
...
     }
 
+    // checks if there is space in the queue for a new record
     pub fn can_add_record(&self, payload_size: usize) -> bool {
         // Desktop does the cur_bytes check as exclusive, but we shouldn't see any servers that
         // don't have https://github.com/mozilla-services/server-syncstorage/issues/73
         self.cur_records < self.max_records && self.cur_bytes + payload_size <= self.max_bytes
     }
 
+    // returns a boolean based on the size of a record versus the max space in the queue
```
Comment (ml8) [inline components/sync15/src/request.rs]: Returns true iff this record is larger than the queue's maximum size.

File: components/sync15/src/request.rs [hunk 4]
```diff
@@ -34,21 +37,25 @@ impl LimitTracker {
         }
     }
 
+    // resets the limit count
     pub fn clear(&mut self) {
         self.cur_records = 0;
         self.cur_bytes = 0;
     }
 
+    // checks if there is space in the queue for a new record
```
Comment (ml8) [inline components/sync15/src/request.rs]: Returns true iff this record can be added to the queue.

File: components/sync15/src/request.rs [hunk 5]
```diff
@@ -24,7 +24,10 @@ struct LimitTracker {
     cur_records: usize,
 }
 
+// The implementation for the limit, allowing for a new limit,
+// to reset the limit, and keep track of records count in the queue
 impl LimitTracker {
+    // creates a new limit
```
Comment (ml8) [inline components/sync15/src/request.rs]: Creates a new, empty, LimitTracker with the given queue size in bytes and records.

File: components/sync15/src/request.rs [hunk 6]
```diff
@@ -34,21 +37,25 @@ impl LimitTracker {
         }
     }
 
+    // resets the limit count
```
Comment (ml8) [inline components/sync15/src/request.rs]: Clears the current queue and resets the tracker's state.

File: components/sync15/src/request.rs [hunk 7]
```diff
@@ -34,21 +37,25 @@ impl LimitTracker {
...
         // don't have https://github.com/mozilla-services/server-syncstorage/issues/73
         self.cur_records < self.max_records && self.cur_bytes + payload_size <= self.max_bytes
     }
 
+    // returns a boolean based on the size of a record versus the max space in the queue
     pub fn can_never_add(&self, record_size: usize) -> bool {
         record_size >= self.max_bytes
     }
 
+    // updates the count after a record is added
```
Comment (ml8) [inline components/sync15/src/request.rs]: Adds a record to the queue.

File: components/sync15/src/request.rs (hunk 5 above)
Comment (matthewlang) [inline components/sync15/src/request.rs]: Turn all of these into [doc comments](https://doc.rust-lang.org/rust-by-example/meta/doc.html).

File: components/sync15/src/request.rs [hunk 8]
```diff
@@ -98,6 +105,7 @@ fn default_max_record_payload_bytes() -> usize {
     256 * 1024
 }
 
+// Creates a default instance for the info configuration
```
Comment (ml8) [inline components/sync15/src/request.rs]: rm: not really necessary; the Default trait is well-known.

File: components/sync15/src/request.rs [hunk 9]
```diff
@@ -34,21 +37,25 @@ impl LimitTracker {
...
+    // checks if there is space in the queue for a new record
     pub fn can_add_record(&self, payload_size: usize) -> bool {
         // Desktop does the cur_bytes check as exclusive, but we shouldn't see any servers that
         // don't have https://github.com/mozilla-services/server-syncstorage/issues/73
         self.cur_records < self.max_records && self.cur_bytes + payload_size <= self.max_bytes
     }
 
+    // returns a boolean based on the size of a record versus the max space in the queue
     pub fn can_never_add(&self, record_size: usize) -> bool {
         record_size >= self.max_bytes
```
Comment (ml8) [inline components/sync15/src/request.rs]: Orthogonal to this change... Should this be >? Line 50 seems in conflict.

File: components/sync15/src/request.rs (hunk 5 above)
Comment (ml8) [inline components/sync15/src/request.rs]: Make these [doc comments](https://doc.rust-lang.org/rust-by-example/meta/doc.html).

File: components/sync15/src/request.rs [hunk 10]
```diff
@@ -129,6 +137,7 @@ impl Deref for InfoCollections {
     }
 }
 
+// A struct to upload the result of a batch request
```
Comment (ml8) [inline components/sync15/src/request.rs]: Documenting this is good since it is public, but make this a [doc comment](https://doc.rust-lang.org/rust-by-example/meta/doc.html). 

"A struct" is redundant. Consider:

`/// The result of a batch upload, containing results for successful and failed updates.`

File: components/sync15/src/request.rs [hunk 11]
```diff
@@ -142,13 +151,14 @@ pub struct UploadResult {
 
 pub type PostResponse = Sync15ClientResponse<UploadResult>;
 
+//REMOVE BatchState Unnsupported for issue 1144
```
Comment (ml8) [inline components/sync15/src/request.rs]: Rm this comment. The enum itself does make sense: uploading batches is stateful, so we do need to know whether we are in a batch or not.

Consider renaming `NoBatch` to `NewBatch`?

File: components/sync15/src/request.rs [hunk 12]
```diff
@@ -142,13 +151,14 @@ pub struct UploadResult {
...
 #[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
 pub enum BatchState {
-    Unsupported,
     NoBatch,
     InBatch(String),
 }
 
+// The struct holding all the options for the queue
```
Comment (ml8) [inline components/sync15/src/request.rs]: Here and elsewhere, "the struct" is redundant. Also, if the comment only contains the information that is in the name of the type, then it isn't a useful comment.

Also a `PostQueue` has rich behavior, this isn't just options--this is the struct that does the `add`ing of records and `flush`ing of batches of records to the service, so the comment is actually misleading.

This should be documented though, but please be more descriptive of _what_ it does.

File: components/sync15/src/request.rs [hunk 13]
```diff
@@ -162,6 +172,7 @@ pub struct PostQueue<Post, OnResponse> {
     last_modified: ServerTimestamp,
 }
 
+// How to deal with a batch
```
Comment (ml8) [inline components/sync15/src/request.rs]: This comment doesn't help me understand anything.

What does a type that implements `BatchPoster` do?

File: components/sync15/src/request.rs [hunk 14]
```diff
@@ -182,6 +193,7 @@ pub trait PostResponseHandler {
     fn handle_response(&mut self, r: PostResponse, mid_batch: bool) -> Result<()>;
 }
 
+// A struct defining responses for the batch poster
```
Comment (ml8) [inline components/sync15/src/request.rs]: rm or update

File: components/sync15/src/request.rs [hunk 15]
```diff
@@ -191,6 +203,7 @@ pub(crate) struct NormalResponseHandler {
     pub pending_success: Vec<Guid>,
 }
 
+// The implementation of handling responses for the batcher
```
Comment (ml8) [inline components/sync15/src/request.rs]: rm

File: components/sync15/src/request.rs [hunk 16]
```diff
@@ -203,6 +216,8 @@ impl NormalResponseHandler {
     }
 }
 
+// This implementation puts the responses into arrays so they can
```
Comment (ml8) [inline components/sync15/src/request.rs]: rm. I do agree that the actual function itself could be better commented, but this comment belongs _inside_ the function, since it's not documentation for users of this function.

File: components/sync15/src/request.rs [hunk 17]
```diff
@@ -227,6 +242,7 @@ impl PostResponseHandler for NormalResponseHandler {
     }
 }
 
+// The main function that posts the batch queue
```
Comment (ml8) [inline components/sync15/src/request.rs:236]: rm

File: components/sync15/src/request.rs [hunk 18]
```diff
@@ -251,17 +267,21 @@ where
         }
     }
 
+    // UPDATED TO NOT ALLOW UNSUPPORTED BATCHSTATE
```
Comment (ml8) [inline components/sync15/src/request.rs]: rm

File: components/sync15/src/request.rs [hunk 19]
```diff
@@ -251,17 +267,21 @@ where
...
+    // Determines if there is a batch
```
Comment (ml8) [inline components/sync15/src/request.rs]: Returns true iff we are in the middle of uploading a batch.

File: components/sync15/src/request.rs [hunk 20]
```diff
@@ -251,17 +267,21 @@ where
...
     #[inline]
     fn in_batch(&self) -> bool {
         match &self.batch {
-            BatchState::Unsupported | BatchState::NoBatch => false,
+            BatchState::NoBatch => false,
             _ => true,
         }
     }
 
+    // This puts an item into the batch queue
```
Comment (ml8) [inline components/sync15/src/request.rs]: Make a doc comment. Describe what it returns and why.

File: components/sync15/src/request.rs [hunk 21]
```diff
@@ -251,17 +267,21 @@ where
...
     pub fn enqueue(&mut self, record: &EncryptedBso) -> Result<bool> {
         let payload_length = record.payload.serialized_len();
 
+        // Checks on payload_lengths and bytes
```
Comment (ml8) [inline components/sync15/src/request.rs]: Be more descriptive. What is the intent of this code? What will help me understand what its intent is? 

Consider the difference between
`// Checks on payload_lengths and bytes`
and
`// Return early if the record can never be submitted with the current server limits.`.

File: components/sync15/src/request.rs [hunk 22]
```diff
@@ -301,6 +321,8 @@ where
         // The + 1 is only relevant for the final record, which will have a trailing ']'.
         let item_len = item_end - item_start + 1;
 
+        // Catches errors if the item being put into the batch
```
Comment (ml8) [inline components/sync15/src/request.rs]: Same here. This code is actually handling the error where the record is too large to ever be updated.

The check on line 284 is the result of `payload_length`, which is supposed to provide an overestimate. This check is for the actual serialized length, which may be less than `payload_length`.

`// Check if the actual serialized length of the record is too long to ever be submitted with current server limits.`

File: components/sync15/src/request.rs [hunk 23]
```diff
@@ -314,6 +336,8 @@ where
         let can_batch_record = self.batch_limits.can_add_record(payload_length);
         let can_send_record = self.queued.len() < self.max_request_bytes;
 
+        // If any of the limits have been breached,
```
Comment (ml8) [inline components/sync15/src/request.rs]: This is great comment!

Reword: `If any of the queue limits have been exceeded by enqueueing the record, remove the record, flush the queue, and re-enqueue the record.`

File: components/sync15/src/request.rs [hunk 24]
```diff
@@ -331,12 +355,15 @@ where
             serde_json::to_writer(&mut self.queued, &record).unwrap();
         }
 
+        // If all the error checks pass,
```
Comment (ml8) [inline components/sync15/src/request.rs]: `Add the record to the queue tracker.`

File: components/sync15/src/request.rs [hunk 25]
```diff
@@ -331,12 +355,15 @@ where
...
+        // update the current limits
         self.post_limits.record_added(payload_length);
         self.batch_limits.record_added(payload_length);
 
         Ok(true)
     }
 
+    // This flushes/clears certain parts of the batch based on parameters
```
Comment (ml8) [inline components/sync15/src/request.rs]: Also document what `want_commit` means, and what can be returned.

I think `Flushes the current queue` is a better description.

File: components/sync15/src/request.rs [hunk 26]
```diff
@@ -375,13 +400,16 @@ where
 
         self.queued.truncate(0);
 
-        if want_commit || self.batch == BatchState::Unsupported {
+        //Updated to take away unsupported batchstate
```
Comment (ml8) [inline components/sync15/src/request.rs]: rm

File: components/sync15/src/request.rs (hunk 19 above)
Comment (ml8) [reply components/sync15/src/request.rs]: This function is now essentially useless, since there are only two states: a new batch or in the middle of a batch. It can be removed.

File: components/sync15/src/request.rs [hunk 27]
```diff
@@ -349,8 +376,6 @@ where
 
         self.queued.push(b']');
         let batch_id = match &self.batch {
```
Comment (ml8) [inline components/sync15/src/request.rs]: Extract this into a function that returns a String.

```
#[inline]
fn batch_id(&self) -> String {
  // Return current batch id or "true", if not already in a batch.
  match &self.batch {
    BatchState::NoBatch => "true".into(),
    BatchState::InBatch(ref s) => s.clone(),
  }
}

Comment (ml8) [reply components/sync15/src/request.rs]: Then, you can get rid of `batch_id.is_some()` for `is_commit` and change the call to `post` to do `Some(self.batch_id())` instead of `batch_id`.

File: components/sync15/src/request.rs (hunk 26 above)
Comment (matthewlang) [reply components/sync15/src/request.rs]: Stopping here for the time being; I will pick up later.

File: components/sync15/src/request.rs [hunk 28]
```diff
@@ -396,7 +424,7 @@ where
             }
         };
 
-        if want_commit || self.batch == BatchState::Unsupported {
+        if want_commit{
```
Comment (matthewlang) [inline components/sync15/src/request.rs]: fix spacing

File: components/sync15/src/request.rs [hunk 29]
```diff
@@ -375,13 +400,16 @@ where
...
-        if want_commit || self.batch == BatchState::Unsupported {
+        //Updated to take away unsupported batchstate
+        if want_commit{
             self.batch_limits.clear();
         }
         self.post_limits.clear();
 
         let resp = resp_or_error?;
 
+        // A match case for the response from the client,
```
Comment (ml8) [inline components/sync15/src/request.rs]: This comment is confusing.

```
// Update last_modified and batch_id on success, or fail.
```

File: components/sync15/src/request.rs (hunk 28 above)
Comment (ml8) [inline components/sync15/src/request.rs]: Move this into the following if statement, since now there's no condition on batch state.

File: components/sync15/src/request.rs [hunk 30]
```diff
@@ -415,7 +444,8 @@ where
                 .into());
             }
             self.last_modified = last_modified;
-            self.batch = BatchState::Unsupported;
+            //Changed BatchState from Unsupported to NoBatch. Does this work?
```
Comment (ml8) [inline components/sync15/src/request.rs]: This is now a weird mismatch in state. I would say we should return the error directly, as this is the case when the batch id mismatches.

Presumably this would not be a code path that would be exercised.

File: components/sync15/src/request.rs [hunk 31]
```diff
@@ -429,11 +459,8 @@ where
             })?
             .clone();
 
+        // Error handling for a batch failing mid batch
```
Comment (ml8) [inline components/sync15/src/request.rs]: This should now be an `if-let`:

```
// If we are in a batch, fail if there is an id mismatch.
if let BatchState::InBatch(ref cur_id) = self.batch {
  ...
}

File: components/sync15/src/request.rs [hunk 32]
```diff
@@ -455,13 +482,15 @@ where
     }
 }
 
+// A struct used for each piece of information uploaded
```
Comment (ml8) [inline components/sync15/src/request.rs]: Again struct is redundant, also 'piece' implies singular. This comment doesn't add value for me.

File: components/sync15/src/request.rs [hunk 33]
```diff
@@ -455,13 +482,15 @@ where
...
 #[derive(Clone)]
 pub struct UploadInfo {
     pub successful_ids: Vec<Guid>,
     pub failed_ids: Vec<Guid>,
     pub modified_timestamp: ServerTimestamp,
 }
 
+// The response of an uploaded queue is recorded here
```
Comment (ml8) [inline components/sync15/src/request.rs]: Similar

File: components/sync15/src/request.rs [hunk 34]
```diff
@@ -491,6 +520,7 @@ impl<Poster> PostQueue<Poster, NormalResponseHandler> {
     }
 }
 
+// All the tests for the batching code
```
Comment (ml8) [inline components/sync15/src/request.rs]: rm

File: components/sync15/src/request.rs [hunk 35]
```diff
@@ -500,6 +530,8 @@ mod test {
     use std::collections::VecDeque;
     use std::rc::Rc;
     use url::Url;
+
+    // test that the url is compiled correctly
```
Comment (ml8) [inline components/sync15/src/request.rs]: Can this comment be more specific? Or, alternatively, comment the test itself, since `test_url_building` is as meaningful to me as this comment.

File: components/sync15/src/request.rs [hunk 36]
```diff
@@ -875,14 +915,11 @@ mod test {
         assert_eq!(t.batches[1].records, 1);
         assert_eq!(t.batches[1].bytes, payload_size);
         // We know at this point that the server does not support batching.
-        assert_eq!(t.batches[1].posts[0].batch, None);
-        assert_eq!(t.batches[1].posts[0].commit, false);
-        assert_eq!(
-            t.batches[1].posts[0].body.len(),
-            request_bytes_for_payloads(&[payload_size])
-        );
+        // Removed test that tested for unsupported
```
Comment (ml8) [inline components/sync15/src/request.rs]: rm

File: components/sync15/src/request.rs [hunk 37]
```diff
@@ -1153,6 +1196,8 @@ mod test {
         );
     }
 
+    // Tests running multiple post queues on multiple batches,
+    // tests it specifically changing bytes
```
Comment (ml8) [inline components/sync15/src/request.rs:1150]: Commenting these tests is great, but can you be more specific about what they do, either in the tests themselves, or in these comments. Typically the name of the test itself should be a good indication of what it is testing. Then, comments in the test should describe _how_ and _why_ the test is doing what it's doing.

**Review by ml8 — CHANGES_REQUESTED** [review]
Good start! I have a lot of comments about comments, as well as some code that can be cleaned up.

In general, comments should add to understand, and be succinct and precise. A comment that is essentially a restating of the name of a type isn't that helpful if the type has a good name. Saying how the type should be used, or why the type exists is more useful.

File: components/sync15/src/request.rs [hunk 38]
```diff
@@ -25,6 +25,7 @@ struct LimitTracker {
 }
 
 impl LimitTracker {
+    // creates a new, empty, LimitTracker with the given queue size in bytes and records
```
Comment (ml8) [inline components/sync15/src/request.rs]: Here and below use capitalization

File: components/sync15/src/request.rs [hunk 39]
```diff
@@ -244,24 +251,20 @@ where
...
 
-    #[inline]
-    fn in_batch(&self) -> bool {
-        match &self.batch {
-            BatchState::Unsupported | BatchState::NoBatch => false,
-            _ => true,
-        }
-    }
 
+    /// Returns Ok(true) iff a record is added with no error
```
Comment (ml8) [inline components/sync15/src/request.rs:263]: Describe what the function does. Also say under what circumstances it will return false (if the record can't be added given the current server limits). Also, it can return an error (from line 334, for example).

File: components/sync15/src/request.rs [hunk 40]
```diff
@@ -331,57 +338,61 @@ where
...
 
+        // Add the record to the queue tracker
         self.post_limits.record_added(payload_length);
         self.batch_limits.record_added(payload_length);
 
         Ok(true)
     }
 
+
+    /// Returns batchid
```
Comment (ml8) [inline components/sync15/src/request.rs]: Remove comment or make more precise (e.g., move the comment on line 352 up here).

File: components/sync15/src/request.rs [hunk 41]
```diff
@@ -331,57 +338,61 @@ where
...
+    fn batch_id(&self) -> String {
+        //Return current batch_id or "true", if not already in batch
+        match &self.batch {
+            BatchState::NewBatch => "true".into(),
+            BatchState::InBatch(ref s) => s.clone(),
+        }
+    }
+
+    // Flushes the current queue
+    // want_commit??? TODO
```
Comment (ml8) [inline components/sync15/src/request.rs]: Since this is a public function, this comment should be a doc comment.

Also, get rid of comments to yourself.

File: components/sync15/src/request.rs [hunk 42]
```diff
@@ -331,57 +338,61 @@ where
...
+            BatchState::NewBatch => "true".into(),
+            BatchState::InBatch(ref s) => s.clone(),
+        }
+    }
+
+    // Flushes the current queue
+    // want_commit??? TODO 
+    // Returns OK() if there is no error 
     pub fn flush(&mut self, want_commit: bool) -> Result<()> {
+        // Return if the queue is empty
```
Comment (ml8) [inline components/sync15/src/request.rs]: Unnecessary comment

File: components/sync15/src/request.rs [hunk 43]
```diff
@@ -331,57 +338,61 @@ where
...
-        };
 
         log::info!(
             "Posting {} records of {} bytes",
             self.post_limits.cur_records,
             self.queued.len()
         );
 
-        let is_commit = want_commit && batch_id.is_some();
+        let is_commit = want_commit;
```
Comment (ml8) [inline components/sync15/src/request.rs]: Get rid of `is_commit`, since it's just the same as `want_commit`.

File: components/sync15/src/request.rs (hunk 31 above)
Comment (ml8) [reply components/sync15/src/request.rs]: Use my comment, it's more descriptive.

File: components/sync15/src/request.rs [hunk 44]
```diff
@@ -396,26 +407,23 @@ where
...
         if want_commit {
             log::debug!("Committed batch {:?}", self.batch);
-            self.batch = BatchState::NoBatch;
+            self.batch = BatchState::NewBatch;
             self.on_response.handle_response(resp, false)?;
+            self.last_modified = last_modified;
             return Ok(());
         }
 
+        // Error handling for a failed batch
```
Comment (ml8) [inline components/sync15/src/request.rs]: This is an unnecessary comment. Rm or make more descriptive.

File: components/sync15/src/request.rs [hunk 45]
```diff
@@ -586,6 +591,7 @@ mod test {
         cfg: InfoConfiguration,
     }
 
+    // An implementation that tests posting a batch
```
Comment (ml8) [inline components/sync15/src/request.rs]: rm

File: components/sync15/src/request.rs [hunk 46]
```diff
@@ -558,6 +561,8 @@ mod test {
         records: usize,
     }
 
+    // Test posting json records,
+    // check if the correct errors fire
```
Comment (ml8) [inline components/sync15/src/request.rs]: rm

File: components/sync15/src/request.rs [hunk 47]
```diff
@@ -721,6 +727,7 @@ mod test {
 
     type MockedPostQueue = PostQueue<TestPosterRef, TestPosterRef>;
 
+    // this tests the setup of the postqueue
```
Comment (ml8) [inline components/sync15/src/request.rs]: rm, it also doesn't test anything, it just creates a pq for other tests.

File: components/sync15/src/request.rs [hunk 48]
```diff
@@ -731,6 +738,7 @@ mod test {
         (pq, tester)
     }
 
+    // tests giving a response to posting a successful batch
```
Comment (ml8) [inline components/sync15/src/request.rs]: rm

File: components/sync15/src/request.rs [hunk 49]
```diff
@@ -805,6 +813,7 @@ mod test {
             .sum::<usize>()
     }
 
+    // Runs a series of tests on a postqueue
```
Comment (ml8) [inline components/sync15/src/request.rs]: rm

File: components/sync15/src/request.rs [hunk 50]
```diff
@@ -835,6 +844,8 @@ mod test {
         );
     }
 
+    // Tests maxing out a postqueue and tests that
+    // all the error catching works properly
```
Comment (ml8) [inline components/sync15/src/request.rs]: rm or make more descriptive.

File: components/sync15/src/request.rs [hunk 51]
```diff
@@ -875,14 +886,10 @@ mod test {
...
-        assert_eq!(t.batches[1].posts[0].batch, None);
-        assert_eq!(t.batches[1].posts[0].commit, false);
-        assert_eq!(
-            t.batches[1].posts[0].body.len(),
-            request_bytes_for_payloads(&[payload_size])
-        );
     }
 
+    // Does the same as the previous test, but with records
+    // instead of requests
```
Comment (ml8) [inline components/sync15/src/request.rs]: rm or make more descriptive

File: components/sync15/src/request.rs [hunk 52]
```diff
@@ -921,6 +928,7 @@ mod test {
         );
     }
 
+    // Testing if a single batch works
```
Comment (ml8) [inline components/sync15/src/request.rs]: ditto, here and below.

In general, the comment has to tell me more than this to be useful. Typically, for unit tests, the test name should be descriptive enough in order to convey what the test does, and therefore there is usually less comments on tests.

I don't think these tests need comments. Also, though, compare "Testing if a single batch works" with "Validate that all queued records are posted when flushing succeeds" -- the latter is more precise and describes exactly what this test does.

File: components/sync15/src/request.rs [hunk 53]
```diff
@@ -1242,11 +1257,114 @@ mod test {
             request_bytes_for_payloads(&[100])
         );
     }
+ 
+    #[test]
+    fn test_pq_multi_batch_records_and_bytes() {
```
Comment (ml8) [inline components/sync15/src/request.rs:1242]: I'm not really clear what's being tested here. Can you walk me through what you're validating here?

File: components/sync15/src/request.rs [hunk 54]
```diff
@@ -1242,11 +1257,114 @@ mod test {
...
+        assert_eq!(t.all_posts.len(), 1);
+        assert_eq!(t.batches.len(), 1);
+        assert_eq!(t.batches[0].posts[0].commit, true);
+    }
 
     // TODO: Test
     //
-    // - error cases!!! We don't test our handling of server errors at all!
-    // - mixed bytes/record limits
+    // - edge error cases
```
Comment (ml8) [inline components/sync15/src/request.rs]: Which specifically?

File: components/sync15/src/request.rs [hunk 55]
```diff
@@ -1242,11 +1257,114 @@ mod test {
...
+        assert_eq!(t.batches[1].posts[0].payload_bytes, 200);
+        assert_eq!(t.batches[1].posts[0].commit, false);
+        assert_eq!(
+            t.batches[1].posts[0].body.len(),
+            request_bytes_for_payloads(&[100, 100])
+        );
+    }
+    
+    #[test]
+    fn test_error_cases() {
```
Comment (ml8) [inline components/sync15/src/request.rs:1318]: Similarly here, is this WIP?

**Comment by ml8** [comment]
Took another pass through this. Just a few nits. Are the tests a WIP?

Also, I'm going to add a reading this week on comments, since I'm seeing some common patterns across groups. We'll be reading Chapter 32 (self-documenting code).

File: components/sync15/src/request.rs (hunk 53 above)
Comment (colehendo) [reply components/sync15/src/request.rs:1242]: The two tests above it test record limits for batches and byte limits for batches. This tests essentially combines them and tests record and byte limit cases together--both combinations that should succeed and combinations that should not. This test was requested in the TODO section at the bottom.

File: components/sync15/src/request.rs (hunk 55 above)
Comment (colehendo) [reply components/sync15/src/request.rs:1318]: Yes, this is WIP because each time I go back over the code I find a small warning or error catch that needs to be tested. I don't want to merge the PR until I test all error cases.

File: components/sync15/src/request.rs [hunk 56]
```diff
@@ -144,11 +150,11 @@ pub type PostResponse = Sync15ClientResponse<UploadResult>;
 
 #[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
 pub enum BatchState {
-    Unsupported,
-    NoBatch,
+    NewBatch,
     InBatch(String),
 }
 
+//Adds and flushes batch to the service
```
Comment (ml8) [inline components/sync15/src/request.rs]: nit: missing space

File: components/sync15/src/request.rs (hunk 39 above)
Comment (ml8) [reply components/sync15/src/request.rs:263]: Good, please add a brief sentence about when it can throw an error. :)

File: components/sync15/src/request.rs [hunk 57]
```diff
@@ -331,57 +339,55 @@ where
...
         }
 
+        // Add the record to the queue tracker
         self.post_limits.record_added(payload_length);
         self.batch_limits.record_added(payload_length);
 
         Ok(true)
     }
 
+    //Return current batch_id or "true", if not already in batch
```
Comment (ml8) [inline components/sync15/src/request.rs]: nit: space

File: components/sync15/src/request.rs (hunk 31 above)
Comment (ml8) [reply components/sync15/src/request.rs]: ping

File: components/sync15/src/request.rs (hunk 55 above)
Comment (ml8) [reply components/sync15/src/request.rs:1318]: 👍

**Review by ml8 — COMMENTED** [review]
Code looks good, just a couple of nits! I'll wait on tests. Please re-request my review when the tests are added.

Thanks!

File: components/sync15/src/request.rs [hunk 58]
```diff
@@ -1243,10 +1238,115 @@ mod test {
...
     }
 
-    // TODO: Test
-    //
-    // - error cases!!! We don't test our handling of server errors at all!
-    // - mixed bytes/record limits
-    //
-    // A lot of these have good examples in test_postqueue.js on deskftop sync
+    #[test]
+    fn test_pq_multi_batch_records_and_bytes() {
```
Comment (ml8) [inline components/sync15/src/request.rs:1242]: add comment or remove the comments on other tests

**Review by ml8 — APPROVED** [review]
LGTM! Yay!

Just a couple of things I'd like to see before you send out:
* Please update PR title and description so that it describes the change. The title should be something like "remove support for non-batching servers in..." The description should be a brief description of the change.
* Please remove the things I marked "rm"
* I added one comment that you should either add comments for all tests or none of them :)
